### PR TITLE
Feat/active member composition line graph

### PR DESCRIPTION
**Pull request by Zueis**
[Feature]:
- Implement Active member composition UI.
- Integrate with API for collecting data from the server and show in the graph.
- add the Onboarding UI component.
- add Disengaged members composition UI component.
- add Inactive members UI component.
- responsive components.
- add a tooltip for the description of statistical data component.

### CODE REVIEW

File: src/components/pages/statistics/ActiveMembersComposition.tsx
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/statistics/ActiveMembersComposition.tsx:1]: It's not the important thing to focus on it. style of coding is different for everyone.

File: src/components/pages/statistics/DisengagedMembersComposition.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import { useEffect, useState } from 'react';
+import useAppStore from '../../../store/useStore';
+import LineGraph from '../../global/LineGraph';
+import StatisticalData from './StatisticalData';
+import { FiCalendar } from 'react-icons/fi';
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/statistics/DisengagedMembersComposition.tsx:5]: You have a different config in your vs code. If your config is better please create an issue on the repo and add a new config. So, I can use the same eslint config.

File: src/components/pages/statistics/DisengagedMembersComposition.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    //     };
+    //   }
+    //   return interaction;
+    // });
+
+    // newOptions.series = newSeries;
+    // newOptions.xAxis.categories = interactions.categories;
+
+    // setOptions(newOptions);
+
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/statistics/DisengagedMembersComposition.tsx:109]: this component implements but not finished as we are waiting for APIs. So, don't focus on this component until we get APIs for this one.

File: src/components/pages/pageIndex/ActiveMemberComposition.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import StatisticalData from '../statistics/StatisticalData';
+import useAppStore from '../../../store/useStore';
+import CustomButton from '../../global/CustomButton';
+import { useRouter } from 'next/router';
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/pageIndex/ActiveMemberComposition.tsx:5]: like top comment

File: src/components/pages/pageIndex/ActiveMemberComposition.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+import { StorageService } from '../../../services/StorageService';
+import { IUser } from '../../../utils/types';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/pageIndex/ActiveMemberComposition.tsx:8]: like top one

File: src/components/pages/pageIndex/ActiveMemberComposition.tsx [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+import RangeSelect from '../../global/RangeSelect';
+import { FiCalendar } from 'react-icons/fi';
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/pageIndex/ActiveMemberComposition.tsx:10]: like top one

File: src/components/pages/statistics/Onboarding.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    //     };
+    //   }
+    //   return interaction;
+    // });
+
+    // newOptions.series = newSeries;
+    // newOptions.xAxis.categories = interactions.categories;
+
+    // setOptions(newOptions);
+
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/statistics/Onboarding.tsx:81]: not finished yet. we need api

File: src/pages/statistics.tsx [hunk 7]
```diff
@@ -1,7 +1,11 @@
 import { useEffect, useState } from 'react';
 import CustomTab from '../components/global/CustomTab';
 import { defaultLayout } from '../layouts/defaultLayout';
+import ActiveMembersComposition from '../components/pages/statistics/ActiveMembersComposition';
+// import Onboarding from '../components/pages/statistics/Onboarding';
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/pages/statistics.tsx:5]: need api to finish

File: src/components/pages/statistics/DisengagedMembersComposition.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+  {
+    title: '1Y',
+    value: 5,
+  },
+];
+
+export default function ActiveMembersComposition({
+  activePeriod,
+  handleDateRange,
+}: any) {
```
[the remark this answers is not in the document]
Comment (Zueis) [reply src/components/pages/statistics/DisengagedMembersComposition.tsx]: fixed

**Review by webmiraclepro — APPROVED** [review]

**Pull request merged**